app/src/main/res/layout/view_attachment_details.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  ~ Infomaniak Mail - Android
+  ~ Copyright (C) 2023 Infomaniak Network SA
+  ~
+  ~ This program is free software: you can redistribute it and/or modify
+  ~ it under the terms of the GNU General Public License as published by
+  ~ the Free Software Foundation, either version 3 of the License, or
+  ~ (at your option) any later version.
+  ~
+  ~ This program is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  ~ GNU General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU General Public License
+  ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content">
+
+    <ImageView
+        android:id="@+id/icon"
+        android:layout_width="@dimen/standardIconSize"
+        android:layout_height="@dimen/standardIconSize"
+        android:layout_marginHorizontal="@dimen/marginStandardSmall"
+        android:layout_marginVertical="@dimen/marginStandardVerySmall"
+        android:importantForAccessibility="no"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toStartOf="@id/textsLayout"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:src="@drawable/ic_file_pdf" />
+
+    <LinearLayout
+        android:id="@+id/textsLayout"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="@dimen/marginStandardVerySmall"
+        android:layout_marginVertical="@dimen/marginStandardVerySmall"
+        android:orientation="vertical"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@id/icon"
+        app:layout_goneMarginEnd="@dimen/marginStandardSmall">
+
+        <TextView
+            android:id="@+id/fileName"
+            style="@style/BodySmall"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:ellipsize="middle"
+            android:lines="1"
+            android:maxWidth="220dp"
+            tools:text="billet-055.pdf" />
+
+        <TextView
+            android:id="@+id/fileSize"
+            style="@style/Label.Secondary"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            tools:text="36 ko" />
+    </LinearLayout>
```

Comment:
It's not possible to get rid of the `LinearLayout` ? Usually, we don't need another level of intrication when using a `ConstraintLayout`.

Reply:
It is possible to get rid of it, but I kept it because it allows to "factorize" the margins and the dimensions of the two separate textViews
With the constraint layout I thought that it would be easy to forget to update a margin or padding in one or the other Views.
It also make the entire file a lot easier to read
I think the performance tradeoff here should be really low as the view is very simple

What do you think ?

Reply:
I just realize that this is not the latest version of this file
this was a work-in-progress that was an extraction in another file of the existing view, without refactoring